Workflow/Sources/WorkflowAction.swift

```diff
@@ -71,16 +81,35 @@ public struct AnyWorkflowAction<WorkflowType: Workflow>: WorkflowAction {
         self.init(closureAction: closureAction)
     }
 
+    /// Creates a type-erased workflow action with the given `apply` implementation.
+    ///
+    /// - Parameter apply: the apply function for the resulting action.
+    @_disfavoredOverload
+    public init(
+        _ apply: @escaping (inout WorkflowType.State) -> WorkflowType.Output?,
+        fileID: StaticString = #fileID,
+        line: UInt = #line
+    ) {
+        self.init(
+            { state, _ in apply(&state) },
+            fileID: fileID,
+            line: line
+        )
+    }
+
```

Comment:
This means we don't have to go update a bunch of existing callsites that use the current `apply` API right?